test/pylib/manager_client.py

```diff
@@ -229,6 +229,41 @@ async def server_add(self, replace_cfg: Optional[ReplaceConfig] = None,
             await self.driver_connect()
         return s_info
 
+    async def servers_add(self, servers_num: int = 1,
+                          cmdline: Optional[List[str]] = None,
+                          config: Optional[dict[str, Any]] = None,
+                          property_file: Optional[dict[str, Any]] = None,
+                          start: bool = True,
+                          expected_error: Optional[str] = None) -> [ServerInfo]:
+        """Add new servers concurrently"""
+        assert servers_num > 0, f"servers_add: cannot add {servers_num} servers, servers_num must be positive"
+
+        try:
+            data = self._create_server_add_data(None, cmdline, config, property_file, start, expected_error)
+            data['servers_num'] = servers_num
+            server_infos = await self.client.put_json("/cluster/addservers", data, response_type="json",
+                                                      timeout=ScyllaServer.TOPOLOGY_TIMEOUT * servers_num)
+        except Exception as exc:
+            raise Exception("Failed to add servers") from exc
```

Comment:
Please verify if this preserves the original contents of `exc` (e.g. by modifying Scylla to throw on boot and seeing what happens). I recall having some problems with this `from exc` thing, where details about the error were lost and made debugging failures from CI harder; formatting the original as string and including it into the new exception's string gave me better results (maybe we could/should do both?)

Reply:
It doesn't preserve the original contents of `exc` and I think it works as intended. This `from exc` thing is just to indicate in code that the raised exception is a direct consequence of the catched exception.

What do you mean by "formatting the original as string"? If I simply print `exc`, I get:
```
        except Exception as exc:
>           raise Exception(f"Failed to add servers: {exc}") from exc
E           Exception: Failed to add servers: HTTP error 500, uri: http+unix://api/cluster/addservers, params: None, json: {'start': True, 'servers_num': 3}, body:
E           SOME ERROR
```
I don't think we want to show this whole HTTP info. We already log it in `RESTClient._fetch`. I don't think logging it again makes a lot of sense.

Another option is to catch `HTTPError` instead and print `exc.message`. But again, it will be logged for the second time (maybe in a nicer way though). Also, it might not be correct if, for example, a `RuntimeError` raise is also possible.

Maybe it's better to leave it as it is? If you still prefer printing `exc`, please specify how. Then, what should I do with the rest of the `from exc` things present in the code? I could change all of them in the same way in a follow-up.

Reply:
I meant `str(exc)` (which is also what `f"... {exc} ..."` does IIRC).

The `SOME ERROR` part of the message contains the root cause of the error and it should definitely be shown when test fails. The HTTP thing is not necessary I agree, but it also doesn't really hurt.

If it's as you said, that we're already printing this error earlier, then it's fine. What's important is that we can easily find it in the logs.